compiler/rustc_hir/src/hir.rs

```diff
@@ -198,14 +199,32 @@ impl Path<'_> {
     }
 }
 
+#[derive(Debug, Clone, Copy, HashStable_Generic)]
+pub enum PathSegmentRes {
+    Res(Res),
+    LangItem(DefKind, LangItem),
+}
+
+impl PathSegmentRes {
+    pub fn get(&self, lang_items: &LanguageItems) -> Res {
```

Comment:
This should just take `TyCtxt`

Reply:
Actually no, see my comment below about just using `Res::Def` by itself.

Reply:
`TyCtxt` is defined in `rustc_middle`, which `rustc_hir` doesn't depend on.